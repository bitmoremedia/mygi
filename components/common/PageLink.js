import React from 'react';
import { Link } from 'react-router';
import { prefixLink } from 'gatsby-helpers';

const PageLink = ({ to, children, className }) => {
  // add a trailing slash if there is not one (to support SPA)
  let linkTo = to;
  if (to.substr(to.length-1) !== '/') {
    linkTo = to + '/';
  }
  const prefixedLink = prefixLink(linkTo);
  return (
    <Link className={className} to={prefixedLink}>
      {children}
    </Link>
  );
};

export default PageLink;
