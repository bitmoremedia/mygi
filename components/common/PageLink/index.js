import React, { PropTypes } from 'react'
import { Link } from 'react-router'

// gatsby workaround
// the below looks weird, but it means that we can use this component
// outside of the gatsby build run time
// which is required when running a standalone test for this component
let prefixLink = linkTo => linkTo
try {
  const gatsbyHelpers = require('gatsby-helpers') // eslint-disable-line
  prefixLink = gatsbyHelpers.prefixLink
} catch (e) {
  console.log('gatsby=helpers ignored') // eslint-disable-line
}

const PageLink = ({ to, children, className }) => {
  // add a trailing slash if there is not one (to support SPA)
  let linkTo = to
  if (to.substr(to.length - 1) !== '/') {
    linkTo = `${to}/`
  }
  const prefixedLink = prefixLink(linkTo)
  return (
    <Link className={className} to={prefixedLink}>
      {children}
    </Link>
  )
}

PageLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
}

PageLink.defaultProps = {
  className: '',
}

export default PageLink
