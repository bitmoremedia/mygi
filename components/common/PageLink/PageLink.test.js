import React from 'react'
import { shallow } from 'enzyme'
import PageLink from './index'

import { Link } from 'react-router'

test('renders a react router Link component', ()=>{
  const wrapper = shallow(
    <PageLink
      to="somewhere"
    >
      A page link
    </PageLink>
  )
  expect(wrapper.find(Link)).toHaveLength(1)
})

test('the router Link component has a trailing slash added if ommited', ()=>{
  const wrapper = shallow(
    <PageLink
      to="somewhere"
    >
      A page link
    </PageLink>
  )
  const RouterLink = wrapper.find(Link);
  expect(RouterLink.props().to).toEqual('somewhere/')
})

test('the router Link component includes the trailing slash if one is provided', ()=>{
  const wrapper = shallow(
    <PageLink
      to="somewhere-else/"
    >
      A page link
    </PageLink>
  )
  const RouterLink = wrapper.find(Link);
  expect(RouterLink.props().to).toEqual('somewhere-else/')
})

test('the router Link component is passed any provided className', ()=>{
  const wrapper = shallow(
    <PageLink
      to="somewhere-else/"
      className="some-class"
    >
      A page link
    </PageLink>
  )
  const RouterLink = wrapper.find(Link);
  expect(RouterLink.props().className).toEqual('some-class')
})

test('the router Link component is passed any provided children', ()=>{
  const wrapper = shallow(
    <PageLink
      to="somewhere-else/"
      className="some-class"
    >
      A page link
    </PageLink>
  )
  const RouterLink = wrapper.find(Link);
  expect(RouterLink.props().children).toEqual('A page link')
})
