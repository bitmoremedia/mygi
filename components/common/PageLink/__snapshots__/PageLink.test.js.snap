// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot render 1`] = `
<a
  className=""
  onClick={[Function]}
  style={Object {}}
>
  A page link
</a>
`;
